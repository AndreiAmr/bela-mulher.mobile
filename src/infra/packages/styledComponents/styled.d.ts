import 'styled-components/native';

export interface CustomThemeProps {
  borderRadius: string;

  colors: {
    main: string;
    secondary: string;
    gray: string;
    onboarding_background: string;
    purple: string;
  };
}

declare module 'styled-components/native' {
  export interface DefaultTheme extends CustomThemeProps {}
}
