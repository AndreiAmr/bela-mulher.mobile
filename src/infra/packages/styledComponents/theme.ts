import { CustomThemeProps } from './styled';

export const styledTheme: CustomThemeProps = {
  borderRadius: '20px',
  colors: {
    main: '#ff0',
    secondary: 'bรง',
  },
};
