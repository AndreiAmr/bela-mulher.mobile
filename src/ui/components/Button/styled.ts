import { css, styled } from 'styled-components/native';
import { styledTheme } from '../../../infra/packages/styledComponents/theme';
import { RFValue } from 'react-native-responsive-fontsize';

interface ContainerProps {
  color?: keyof typeof styledTheme.colors;
}

export const Container = styled.TouchableOpacity<ContainerProps>`
  background: ${({ theme, color }) =>
    color ? theme.colors[color] : theme.colors.main};

  border-radius: ${RFValue(10)}px;
  height: 50px;
  width: 100%;

  align-items: center;
  justify-content: center;

  ${({ disabled }) =>
    disabled &&
    css`
      opacity: 0.5;
    `}
`;

export const Label = styled.Text`
  color: #fff;
  font-size: ${RFValue(16)}px;
  font-weight: medium;
`;
