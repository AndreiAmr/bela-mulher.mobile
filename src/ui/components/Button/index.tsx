import React from 'react';

import * as S from './styled';
import { styledTheme } from '../../../infra/packages/styledComponents/theme';
import { ActivityIndicator } from 'react-native';

interface ButtonProps {
  label: string;
  onPress(): void;
  color?: keyof typeof styledTheme.colors;

  testId?: string;
  disabled?: boolean;
  isLoading?: boolean;
}

const Button = ({
  label,
  onPress,
  color,
  testId,
  disabled,
  isLoading,
}: ButtonProps) => {
  return (
    <S.Container
      onPress={onPress}
      color={color}
      testID={testId}
      disabled={disabled || isLoading}>
      {isLoading ? (
        <ActivityIndicator size="large" color={'#fff'} />
      ) : (
        <S.Label>{label}</S.Label>
      )}
    </S.Container>
  );
};

export default Button;
