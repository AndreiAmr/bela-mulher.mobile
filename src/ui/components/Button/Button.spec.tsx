import 'react-native';
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Button from '.';
import { ThemeProvider } from 'styled-components/native';
import { styledTheme } from '../../../infra/packages/styledComponents/theme';

jest.mock('react-native-responsive-fontsize', () => ({
  RFValue: jest.fn((prop: number) => prop),
}));

describe('Button Component', () => {
  it('Should render correctly', () => {
    const pressFN = () => {
      console.log('pressed');
    };

    const { toJSON } = render(
      <ThemeProvider theme={styledTheme}>
        <Button
          color="gray"
          label="Click me"
          onPress={pressFN}
          testId="button-testid"
        />
      </ThemeProvider>,
    );

    const element = screen.getByTestId('button-testid');

    expect(element).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
