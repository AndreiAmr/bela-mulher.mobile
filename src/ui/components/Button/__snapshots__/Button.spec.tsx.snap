// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6A6A6A",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="button-testid"
>
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 12,
        "fontWeight": "medium",
      }
    }
  >
    Click me
     salve
  </Text>
</View>
`;
